File: giphy/src/main/java/com/valgoodchat/giphy/service/GiphyService.java
package com.valgoodchat.giphy.service;

import com.valgoodchat.giphy.model.GiphyModel;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Configuration used to get GIF from giphy API
 */

public interface GiphyService {
    String SERVICE_ENDPOINT = "https://api.giphy.com/v1/";

    @GET("stickers/search?")
    Observable<GiphyModel> searchGif(@Query("q") String keyword, @Query("limit") String limit,
                                     @Query("offset") String offset, @Query("lang") String lang);

    @GET("gifs/trending?")
    Observable<GiphyModel> trending(@Query("limit") String limit, @Query("offset") String offset);
}

File: giphy/src/main/java/com/valgoodchat/giphy/service/GiphyService.kt
package com.valgoodchat.giphy.service

import com.valgoodchat.giphy.model.GiphyModel

import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

/**
 * Configuration used to get GIF from giphy API
 */

interface GiphyService {

    @GET("stickers/search?")
    fun searchGif(@Query("q") keyword: String, @Query("limit") limit: String,
                  @Query("offset") offset: String, @Query("lang") lang: String): Observable<GiphyModel>

    @GET("gifs/trending?")
    fun trending(@Query("limit") limit: String, @Query("offset") offset: String): Observable<GiphyModel>
}
